pkg/cmd/agent/root.go

```diff
@@ -65,6 +65,11 @@ var rootCmd = &cobra.Command{
 			return errors.New("configuration file must specify auth-token")
 		}
 
+		// Fill provider config
+		c.S3.AutoFill()
```

Comment:
I'm thinking about this part in the context of adding AutoFill to the `/node_info` endpoint.
If the intention is to fill the region value of config, so that AutoFill() doesn't need to read the metadataAPI again, then it's wrong, because AutoFill() just logs on errors. If the provider is not AWS, then the region will never be set and it always need to query the metadata again.

Reply:
Right, so we can delete the `AutoFill` from `/node_info` endpoint.
Note that we are not really concerned with problems like "what if we are deployed in aws but AutoFill failed because of unrelated reasons?", because in such cases rclone server would be incorrectly initialized for the s3 backend and it wouldn't work anyway.